packages/zosjobs/src/SearchJobs.ts

```diff
@@ -0,0 +1,162 @@
+
+            for (const spoolFile of spoolFiles) {
+                fileCount++;
+                let startingLine = 0;
+                let lineCount = 1;
+
+                //If the max number of files have been search then end the search.
+                if(fileCount > fileLimit){
+                    Logger.getAppLogger().debug("searchJobs() - File limit reached");
+                    return replyBuffer;
+                }
+
+                while(startingLine >= 0){
+                    const spoolContent = await this.searchSpoolContentCommon(session, searchString, searchRegex,
+                        caseInsensitive, spoolFile, startingLine);
+                    if(spoolContent.dataString.length > 0){
+                        if(startingLine == 0){
+                            if (spoolFile.procstep != null && spoolFile.procstep.length > 0) {
+                                replyBuffer = replyBuffer + "Job Name: " + job.jobname + " Job Id: " + job.jobid +
+                                    " Spool file: " + spoolFile.ddname + "(ID #" + spoolFile.id + " Step: " +
+                                    spoolFile.stepname + " ProcStep: " + spoolFile.procstep +")" + "\n";
+
+                            } else {
+                                replyBuffer = replyBuffer + "Job Name: " + job.jobname + " Job Id: " + job.jobid +
+                                    " Spool file: " + spoolFile.ddname + "(ID #" + spoolFile.id + " Step: " +
+                                    spoolFile.stepname + ")" + "\n";
+                            }
+                        }
+                        const recordRange:string = spoolContent.response.headers['x-ibm-record-range'];
+                        const lineNumberString:string = recordRange.substring(0, recordRange.indexOf(','));
+                        startingLine = Number(lineNumberString) + 1;
+                        replyBuffer = replyBuffer + " Line " + startingLine + " : " + spoolContent.dataString;
+
+                        // If the search length is exceeded then end the search of this file.
+                        lineCount++;
+                        if(lineCount > searchLimit){
+                            Logger.getAppLogger().debug("searchJobs() - Search limit reached");
+                            startingLine = -1;
+                            replyBuffer = replyBuffer + "\n";
+                        }
+                    }
+                    else{
+                        // If nothing more is found in this file, move on to the next one.
+                        if(startingLine > 0){
+                            replyBuffer = replyBuffer + "\n";
+                        }
+                        startingLine = -1;
+                    }
+                }
+            }
+        }
+
+        return replyBuffer;
+    }
+
+    /**
+     * Get jobs (defaults to the user ID of the session as owner)
```

Comment:
Are the references to 'get' (here and below) left over from some early implementation efforts, or maybe a copy/paste error? I would expect them both to refer to 'search'.

Reply:
They were a copy/paste remnant. Fixed.